src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -628,6 +629,169 @@ JSON example:
+files: `*_asl.json` and `*_aslcontext.tsv`. 
+
+### `*_aslcontext.tsv`
+The `*_aslcontext.tsv` lists the volume_type for each volume in a single column, in the `*_asl.nii[.gz]`, corresponding to the volume_types specified in the table below, based 
+on DICOM Tag (0018,9257) `ASL Context`. Note that the volume_types `control` and  `label` within BIDS only serve to specify the magnetization state of the blood and thus the ASL 
+subtraction order.
+
+| **volume_type** | **Definition**                                                                                                                                                                         |
+|-----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| control         | The control image is acquired in the exact same way as the label image, except that the magnetization of the blood flowing into the imaging region has not been inverted.              |
+| label           | The label image is acquired in the exact same way as the control image, except that the blood magnetization flowing into the imaging region has been inverted.                         |
+| m0scan          | The M0 image is a calibration image, used to estimate the equilibrium magnetization of blood.                                                                                          |
+| deltam          | The deltaM image is a perfusion-weighted image, obtained by the subtraction of `control` - `label`.                                                                                    |
+| cbf             | The cerebral blood flow (CBF) image is produced by dividing the deltaM by the M0, quantified into `mL/100g/min` (See also [doi:10.1002/mrm.25197](https://doi.org/10.1002/mrm.25197)). |
+
+If the `control` and `label` images are not available, their derivative `deltam` should be stored within the `*_asl.nii[.gz]` and specified in the `*_aslcontext.tsv` instead. If 
+the `deltam` is not available, `cbf` should be stored within the `*_asl.nii[.gz]` and specified in the `*_aslcontext.tsv`. When `cbf` is stored within the `*_asl.nii[.gz]`, its 
+units need to be specified in the `*_asl.json` as well. See below examples of these cases, in order of decreasing preference.
+
+#### Case 1: `*_asl.nii[.gz]` consists of volume_types `control`, `label`
+
+In most cases, the ASL timeseries, provided by the scanner, consist of a series of `control` and `label`, and optionally `m0scan`, volumes. In this case, only the `control`, 
+`label`, and optionally `m0scan` volumes should be stored in the `*_asl.json`, and the exact volume_type series should be specified in the `*_aslcontext.tsv`. The optional 
+`deltam` or `cbf` volumes should be stored and specified as derivative. Example of `*_aslcontext.tsv`:
```

Comment:
@effigies Currenlty, there is no specification for derivatives for ASL yet. Adding this would require another round of discussion on which info should be provided for those derivatives. So at this stage, storing these optional DeltaM and CBF images in derivatives is not implemented. This is also planned for a later version. I am not sure how to specify this here?

Reply:
Then could you clarify what you mean by "The optional `deltam` or `cbf` volumes should be stored and specified as derivative."? What I had above was my best guess as to what I thought it meant.

Reply:
@HenkMutsaerts We should discuss this!

Reply:
@effigies I will update by deleting 'The optional deltam or cbf volumes should be stored and specified as derivative' in these cases. The 3 cases will also move to the appendix. Feel free to take a look after change.